overwatch/processing/alarms/collectors.py

```diff
@@ -19,54 +22,155 @@ class Singleton(_Singleton('SingletonMeta', (object,), {})):
+        """ Sends message to specified earlier recipients.
+
+        Args:
+            payload (str): Message to send
+        Return:
+            None.
+        """
+        success = "Emails successfully sent to {recipients}"
+        fail = "EmailDelivery not configured, couldn't send emails"
+
+        if self.recipients is not None:
+            mail = Mail()
+            if 'emailDelivery' in mail.parameters:
+                msg = MIMEMultipart()
+                msg['From'] = mail.user_name
+                msg['To'] = ", ".join(self.recipients)
+                msg['Subject'] = 'Overwatch Alarm'
+                msg.attach(MIMEText(payload, 'plain'))
+                mail.smtp.sendmail(mail.user_name, self.recipients, msg.as_string())
+
+                logger.debug(success.format(recipients=", ".join(self.recipients)))
+            else:
+                logger.debug(fail)
 
 class SlackNotification(Singleton):
-    """ADD DOC"""
+    """Manages sending notifications on Slack.
+
+    Args:
+        alarmsParameters (dict): Parameters read from configuration files
+    Attributes:
+        sc (SlackClient):
+        channel (str): Channel name
+    """
     def __init__(self, alarmsParameters=None):
         if alarmsParameters is not None:
-            self.sc = SlackClient(alarmsParameters["apiToken"])
-            self.channel = alarmsParameters["slackChannel"]
+            self.parameters = alarmsParameters
+            # if 'slack' in self.parameters:
+            try:
+                self.sc = SlackClient(alarmsParameters["slack"]["apiToken"])
+                self.channel = alarmsParameters["slack"]["slackChannel"]
+            except KeyError:
+                logger.debug("Slack not configured")
 
     def __call__(self, alarm):
         self.sendMessage(alarm)
         
     def sendMessage(self, payload):
-        self.sc.api_call('chat.postMessage', channel=self.channel,
-                text=payload, username='Alarms OVERWATCH',
-                icon_emoji=':robot_face:')
+        """ Sends message to specified earlier channel.
+
+        Args:
+            payload (str): Message to send
+        Return:
+            None.
+        """
+        success = "Message successfully sent on Slack channel {channel}"
+        fail = "Slack not configured, couldn't send messages"
+
+        if 'slack' in self.parameters:
+            self.sc.api_call('chat.postMessage', channel=self.channel,
+                    text=payload, username='Alarms OVERWATCH',
+                    icon_emoji=':robot_face:')
+            logger.debug(success.format(channel=self.channel))
+        else:
+            logger.debug(fail)
+
+class AlarmCollector():
+    """
+    Class that collects generated alarms. Collected alarms are grouped and announced to
+    specified receivers.
+
+    Attributes:
+        alarms (list): List of alarms. Each element is a pair [Alarm, str]
```

Comment:
what is str?

Reply:
so it is message, maybe better approach will be keep dict of type Alarm -> List[message:str]